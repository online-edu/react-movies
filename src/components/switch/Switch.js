import React from 'react';
import PropTypes from 'prop-types';

/**
 * Switch component.
 */
const Switch = ({ switchToggle, id, tabIndex, label }) => (
    <div className="custom-control custom-switch">
        <input
            type="checkbox"
            className="custom-control-input"
            aria-label={`${label} Checkbox`}
            id={id}
            name={id}
            onChange={switchToggle}
            tabIndex={tabIndex}
        />
        <label
            className="custom-control-label"
            htmlFor={id}
            aria-label={`${label} filter`}
            aria-describedby={id}
        >
            {label}
        </label>
    </div>
);
/**
 * Switch props types.
 */
Switch.propTypes = {
    /** Id to be assigned to the control. */
    id: PropTypes.number.isRequired,
    /** Display value */
    label: PropTypes.string.isRequired,
    /** Gets called when the user clicks on switch */
    switchToggle: PropTypes.func.isRequired,
    /** Maintain tab index for keyboard navigation. */
    tabIndex: PropTypes.number.isRequired,
};
export default Switch;
