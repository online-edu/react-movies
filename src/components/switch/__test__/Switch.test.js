import React from 'react';
import { shallow, mount } from 'enzyme';
import Switch from '../Switch';

describe('<Switch />', () => {
    const onSwitchClick = jest.fn();
    const wrapper = (
        <Switch
            id="enLangFilter"
            label="Original Language"
            switchToggle={onSwitchClick}
        />
    );

    it('renders correctly', () => {
        const component = shallow(wrapper);
        expect(component).toMatchSnapshot();
    });

    it('contains class name "custom-control, custom-switch" ', () => {
        const component = shallow(wrapper);
        expect(component.find('.custom-control')).toHaveLength(1);
        expect(component.find('.custom-switch')).toHaveLength(1);
    });

    it('allows us to set props', () => {
        const component = mount(wrapper);
        expect(component.props().label).toEqual('Original Language');
        component.setProps({ label: 'Adult' });
        expect(component.props().label).toEqual('Adult');
    });
});
