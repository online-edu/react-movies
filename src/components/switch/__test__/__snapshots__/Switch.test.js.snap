// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Switch
    id="enLangFilter"
    label="Original Language"
    switchToggle={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          aria-label="Original Language Checkbox"
          className="custom-control-input"
          id="enLangFilter"
          name="enLangFilter"
          onChange={[MockFunction]}
          type="checkbox"
        />,
        <label
          aria-describedby="enLangFilter"
          aria-label="Original Language filter"
          className="custom-control-label"
          htmlFor="enLangFilter"
        >
          Original Language
        </label>,
      ],
      "className": "custom-control custom-switch",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Original Language Checkbox",
          "className": "custom-control-input",
          "id": "enLangFilter",
          "name": "enLangFilter",
          "onChange": [MockFunction],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-describedby": "enLangFilter",
          "aria-label": "Original Language filter",
          "children": "Original Language",
          "className": "custom-control-label",
          "htmlFor": "enLangFilter",
        },
        "ref": null,
        "rendered": "Original Language",
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            aria-label="Original Language Checkbox"
            className="custom-control-input"
            id="enLangFilter"
            name="enLangFilter"
            onChange={[MockFunction]}
            type="checkbox"
          />,
          <label
            aria-describedby="enLangFilter"
            aria-label="Original Language filter"
            className="custom-control-label"
            htmlFor="enLangFilter"
          >
            Original Language
          </label>,
        ],
        "className": "custom-control custom-switch",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Original Language Checkbox",
            "className": "custom-control-input",
            "id": "enLangFilter",
            "name": "enLangFilter",
            "onChange": [MockFunction],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-describedby": "enLangFilter",
            "aria-label": "Original Language filter",
            "children": "Original Language",
            "className": "custom-control-label",
            "htmlFor": "enLangFilter",
          },
          "ref": null,
          "rendered": "Original Language",
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
